opentaskspal/src/main/java/org/dmfs/opentaskspal/tasks/SubTaskRelation.java

```diff
@@ -0,0 +1,37 @@
+/*
+ * Copyright 2017 dmfs GmbH
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.dmfs.opentaskspal.tasks;
+
+import org.dmfs.android.contentpal.RowSnapshot;
+import org.dmfs.android.contentpal.rowdata.Composite;
+import org.dmfs.android.contentpal.rowdata.DelegatingRowData;
+import org.dmfs.tasks.contract.TaskContract;
+
+
+/**
+ * @author Gabor Keszthelyi
+ */
+public final class SubTaskRelation extends DelegatingRowData<TaskContract.Properties>
+{
+    public SubTaskRelation(RowSnapshot<TaskContract.Tasks> subTask, RowSnapshot<TaskContract.Tasks> parentTask)
+    {
+        super(new Composite<>(
+                new Relation(subTask, TaskContract.Property.Relation.RelType.PARENT, parentTask),
```

Comment:
it's not necessary to set this on both tasks. In general it's preferred to set the parent relation on the subtask only. But you may have to make sure that any other parent relation with another task is removed first. Although we probably could do that automatically in provider code.

Reply:
Ok, I've modified it to only set one direction of the relation.

What to do with removing other parent relations? Create ticket for implementing in provider then?

Reply:
Yep. Would be nice if we can offload some of the business logic into the provider and make the UI part easier.